torch/utils/data/_utils/collate.py

```diff
@@ -201,6 +202,39 @@ def collate_str_fn(batch, *, collate_fn_map: Optional[Dict[Union[Type, Tuple[Typ
 default_collate_fn_map[bytes] = collate_str_fn
 
 
+def register_default_collate_for(name: Type, collate_fn: Optional[Callable] = None) -> Callable:
+    r"""
+    Register a new collate_fn.
+
+    Args:
+        name: The type of the collate_fn to be applied on.
+        collate_fn: The collate_fn to register.
+
+    Returns:
+        collate_fn: The registered collate_fn.
+
+    Examples:
+        >>> @register_default_collate_for(xxx)
+        ... def collate_xxx_fn(batch: Tuple[PNTensor], collate_fn_map: Optional[Mapping] = None):
+        ...     batch = ...
+        ...     return batch
+        >>> register_default_collate_for(xxx, collate_xxx_fn)
+
+    """
+
+    # register()
+    if collate_fn is not None:
+        default_collate_fn_map[name] = collate_fn
+
+    # @register()
+    @wraps(register_default_collate_for)
+    def register(name, collate_fn):
+        default_collate_fn_map[name] = collate_fn
+        return collate_fn
+
+    return lambda x: register(name, x)
```

Comment:
This is not correct. `wraps` is only useful if you directly return the wrapper function. `lambda` is not working here.
And, you should return a decorator function if `collate_fn` is None.
```py
    def decorator(function):
        def wrapper(collate_fn):
            default_collate_fn_map[name] = collate_fn
            return collate_fn  # Do we need to return collate_fn here? I don't think it's needed.
        return wrapper
    return decorator
```